src/librustc_hir/definitions.rs

```diff
@@ -353,26 +349,38 @@ impl Definitions {
         let parent_hash = DefKey::root_parent_stable_hash(crate_name, crate_disambiguator);
         let def_path_hash = key.compute_stable_hash(parent_hash);
 
-        // Create the definition.
-        let root = LocalDefId { local_def_index: self.table.allocate(key, def_path_hash) };
+        // Create the root definition.
+        let mut table = DefPathTable::default();
+        let root = LocalDefId { local_def_index: table.allocate(key, def_path_hash) };
         assert_eq!(root.local_def_index, CRATE_DEF_INDEX);
-        assert_eq!(self.def_id_to_span.push(rustc_span::DUMMY_SP), root);
+        let mut def_id_to_span = IndexVec::default();
+        assert_eq!(def_id_to_span.push(rustc_span::DUMMY_SP), root);
+
+        Self {
```

Comment:
```suggestion
        Definitions {
```
It's nearly impossible to find places where things are constructed if they are hidden like this.

Reply:
Fixed